src/public-modules/Settings/sagas.js

```diff
@@ -42,40 +38,23 @@ export function* saveSettings(action) {
     ipfsHash: profileImageIpfsHash
   } = values;
 
-  yield put(setPendingWalletConfirm());
-
-  const userAddress = yield select(addressSelector);
-  yield call(getWeb3Client);
-
-  const settings = {
+  const data = {
     name,
     email,
     languages,
     organization,
     skills,
-    social: {
-      personalWebsite: website,
-      github,
-      twitter,
-      linkedin
-      //dribbble: string
-    },
-    profilePhoto: {
-      fileDirectoryHash: profileImageIpfsHash,
-      fileName: fileName
-    }
+    website,
+    github,
+    twitter,
+    linkedin,
+    //dribbble,
+    profileDirectoryHash: profileImageIpfsHash,
+    profileFileName: fileName
   };
 
-  const ipfsHash = yield call(addJSON, settings);
-
-  const { ethProfiles } = yield call(getContractClient);
   try {
-    const txHash = yield call(
-      promisifyContractCall(ethProfiles.addProfile, { from: userAddress }),
-      ipfsHash
-    );
-
-    yield put(setPendingReceipt(txHash));
+    yield call(request, `user/${address}/profile/`, 'POST', { data });
```

Comment:
Is it possible that the server responds with a `4xx` or `5xx` here, and we don't catch it because we don't check the return values?